### PR TITLE
fix: couple FIRE2 variable-cell updates

### PR DESCRIPTION
**Pull request by zubatyuk**
# ALCHEMI Toolkit-Ops Pull Request

## Description

Update FIRE2 variable-cell optimization to run FIRE2 mixing on generalized velocity/force degrees of freedom and then apply a coupled coordinate/cell update. Add atom-count-normalized cell forces, keep `ext_positions` as a deprecated compatibility argument, and update benchmarks, examples, docs, and tests for the coupled FIRE2 cell path.

## Type of Change

- [x] Bug fix (non-breaking change that fixes an issue)
- [x] New feature (non-breaking change that adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Performance improvement
- [x] Documentation update
- [ ] Refactoring (no functional changes)
- [ ] CI/CD or infrastructure change

## Related Issues

None.

## Changes Made

- Add public `fire2_update` support for FIRE2 reduction, adaptive parameter update, and velocity mixing without applying positions.
- Update variable-cell FIRE2 to use coupled coordinate/cell application with atom-count-normalized cell forces and `cell_force_scale`.
- Update FIRE2 variable-cell benchmarks, example usage, and documentation for the coupled coordinate/cell path.
- Add and update FIRE2 tests covering the public update helper, coupled variable-cell references, cell-force scaling, metadata reuse, deprecation handling, and generic extended-path behavior.

## Testing

- [x] Unit tests pass locally (`make pytest`)
- [x] Linting passes (`make lint`)
- [x] New tests added for new functionality meets coverage expectations?

Targeted FIRE2 tests passed. Edited dynamics example and benchmark scripts compile. Minimal FIRE comparison smoke benchmark passed. Staged diff whitespace check passed.

## Checklist

- [x] I have read and understand the [Contributing Guidelines](../CONTRIBUTING.md)
- [ ] I have updated the [CHANGELOG.md](../CHANGELOG.md)
- [x] I have performed a self-review of my code
- [x] I have added docstrings to new functions/classes
- [x] I have updated the documentation (if applicable)

## Additional Notes

None.

> [!TIP]
> This repository uses Greptile, an AI code review service, to help conduct
> pull request reviews. We encourage contributors to read and consider suggestions
> made by Greptile, but note that human maintainers will provide the necessary
> reviews for merging: Greptile's comments are **not** a qualitative judgement
> of your code, nor is it an indication that the PR will be accepted/rejected.
> We encourage the use of emoji reactions to Greptile comments, depending on
> their usefulness and accuracy.

### CODE REVIEW

File: nvalchemiops/torch/fire2.py [hunk 1]
```diff
@@ -544,18 +559,10 @@ def fire2_step_coord_cell(
         alpha0=alpha0,
         tmax=tmax,
         tmin=tmin,
-        maxstep=maxstep,
     )
```
[the remark this answers is not in the document]
Comment (zubatyuk) [reply nvalchemiops/torch/fire2.py:576]: Fixed in ee9b80412bc5dc3790b1ff58dbb9d3dec903a7c5.

File: nvalchemiops/torch/fire2.py [hunk 2]
```diff
@@ -458,6 +483,11 @@ def fire2_step_coord_cell(
         )
     wp_atom_ptr = wp.from_torch(atom_ptr, dtype=wp.int32)
 
+    atom_counts = atom_ptr[1:] - atom_ptr[:-1]
+    cell_force_divisor = atom_counts.to(dtype=dtype).reshape(M, 1, 1) * cell_force_scale
+    cell_force_work = (cell_force.detach() / cell_force_divisor).contiguous()
+    wp_cell_force = wp.from_torch(cell_force_work, dtype=mat_type)
```
[the remark this answers is not in the document]
Comment (zubatyuk) [reply nvalchemiops/torch/fire2.py:502]: Fixed in ee9b80412bc5dc3790b1ff58dbb9d3dec903a7c5.

**Comment by dallasfoster** [comment]
/ok to test [a7eae32](https://github.com/NVIDIA/nvalchemi-toolkit-ops/pull/75/commits/a7eae321acf487a978b600906269381633715b8a)

**Review by dallasfoster — APPROVED** [review]

**Comment by dallasfoster** [comment]
/ok to test [8a678cd](https://github.com/NVIDIA/nvalchemi-toolkit-ops/pull/75/commits/8a678cd5f68008a67a6a9767543fe64f9bc752bb)

**Comment by dallasfoster** [comment]
/ok to test [d4f637f](https://github.com/NVIDIA/nvalchemi-toolkit-ops/pull/75/commits/d4f637f138736230f6b0eacfb54490ca8f1abd6d)

**Comment by laserkelvin** [comment]
/ok to test [c28353b](https://github.com/NVIDIA/nvalchemi-toolkit-ops/pull/75/commits/c28353b00d0f58fcc750bff78febc852729850eb)
